core/src/agent/state.rs

```diff
@@ -30,6 +39,66 @@ pub struct AgentState {
     top_chain_header: Option<ChainHeader>,
 }
 
+impl Serialize for AgentState {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        // 3 is the number of fields in the struct.
+        let mut state = serializer.serialize_struct("AgentState", 3)?;
+        state.serialize_field("keys", &self.chain)?;
+        state.serialize_field("chain_store", &self.chain)?;
```

Comment:
I don't think we need to serialize that one because that is just a wrapper around the CAS. Would mean we could avoid the custom serializer for the filesystemstorage.

Reply:
You can't initiate the agentstate without a CAS. Unless we make the CAS part of it an optional